FastSimulation/Tracking/python/TrajectorySeedProducer_cfi.py

```diff
@@ -10,7 +10,17 @@
     recHitCombinations = cms.InputTag("fastMatchedTrackerRecHitCombinations"),
     layerList = cms.vstring(),
     seedFinderSelector = cms.PSet(
-        measurementTracker = cms.string("")
+        measurementTracker = cms.string(""),
+        layerList = cms.vstring(""),
```

Comment:
How is this `layerList` different from the `layerList` on line 11? From a very quick look on how they're used they seem to be set the same values. If they're indeed always the same, could one of the be dropped?

In addition, I think it would be more clear to remove the empty string arguments, i.e. to use `cms.string()` and `cms.vstring()`.

Reply:
Actually the two layerList parameters are exactly the same. But the one in SeedFinderSelector PSet is required by SeedingLayerSetsBuilder,
https://github.com/angirar/cmssw/blob/1b9eea2212f6b52e5e9a91dad7a6cb16859a44d4/RecoTracker/TkSeedingLayers/src/SeedingLayerSetsBuilder.cc#L246
I will try to think if its possible to remove one of them.

Reply:
Could the higher-level `layerList` be removed and the piece which reads it be changed along
```diff
- iConfig.getParameter<std::vector<std::string>>("layerList")
+ iConfig.getParameter<edm::ParameterSet>("seedFinderSelector").getParameter<std::vector<std::string>>("layerList")
```
?

Please remove the "empty string" arguments from `cms.string` as well in here. And could `layerPairs` be just `cms.vuint32()`?

Reply:
Yes, all done except for when I try to remove the "empty strings" arguments from cms.string I get the following error -
Exception Message:
python encountered the error: <type 'exceptions.TypeError'>
__init__() takes exactly 2 arguments (1 given)

Reply:
That sounds like a problem elsewhere. Could you run `python <your-CMSSW-config>` instead of `cmsRun`? That will give the full stack trace of the failure.

Reply:
This is the output of 'python <config.py>' :
Traceback (most recent call last):
  File "fastsim.py", line 24, in <module>
    process.load('FastSimulation.Configuration.Reconstruction_BefMix_cff')
  File "/cvmfs/cms.cern.ch/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_2_0_pre3/python/FWCore/ParameterSet/Config.py", line 582, in load
    module = __import__(moduleName)
  File "/afs/cern.ch/work/a/arastogi/private/PhaseIUpgrade_Studies/25May2018/CMSSW_10_2_0_pre3/python/FastSimulation/Configuration/Reconstruction_BefMix_cff.py", line 42, in <module>
    from FastSimulation.Tracking.iterativeTk_cff import *
  File "/afs/cern.ch/work/a/arastogi/private/PhaseIUpgrade_Studies/25May2018/CMSSW_10_2_0_pre3/python/FastSimulation/Tracking/iterativeTk_cff.py", line 9, in <module>
    from RecoTracker.IterativeTracking.InitialStep_cff import *
  File "/afs/cern.ch/work/a/arastogi/private/PhaseIUpgrade_Studies/25May2018/CMSSW_10_2_0_pre3/python/RecoTracker/IterativeTracking/InitialStep_cff.py", line 109, in <module>
    import FastSimulation.Tracking.TrajectorySeedProducer_cfi
  File "/afs/cern.ch/work/a/arastogi/private/PhaseIUpgrade_Studies/25May2018/CMSSW_10_2_0_pre3/python/FastSimulation/Tracking/TrajectorySeedProducer_cfi.py", line 13, in <module>
    measurementTracker = cms.string(),
TypeError: __init__() takes exactly 2 arguments (1 given)

Reply:
Right, on a closer look `cms.string()` indeed requires an argument, so the empty strings have to be kept. Sorry for the noise...